### PR TITLE
CVE-2014-1702 and CVE-2014-7908

### PR DESCRIPTION
**Pull request by smcintosh881**
SWEN 331 Krutz

### CODE REVIEW

File: cves/CVE-2014-1702.yml [hunk 1]
```diff
@@ -15,10 +15,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
-bounty:
+description: |
+  This vulnerability is of the free after use nature. When a task is running on a thread,
```
Comment (jonobrien) [inline cves/CVE-2014-1702.yml]: "use after free" ?

File: cves/CVE-2014-1702.yml [hunk 2]
```diff
@@ -15,10 +15,14 @@ description_instructions: |
...
+  and that task ends the thread is being shutdown. This vulnerability could allow an
+  attacker to creat an denial of service. They leverage the improper handling of how
+  these threads are terminated. 
+bounty: 
   date: 2014-03-11 11:00:00.000000000 -04:00
-  amount: 1000.0
+  amount: $1000.0
```
Comment (jonobrien) [inline cves/CVE-2014-1702.yml]: With such a large bounty, can you be more descriptive in how the vulnerability actually impacts the system?

File: cves/CVE-2014-7908.yml [hunk 3]
```diff
@@ -15,25 +15,27 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Integer overflow for multiple media file types. This overflow vulnerability
+  allows attackers to cause a denial of serivce.
```
Comment (jonobrien) [inline cves/CVE-2014-7908.yml]: You might want to describe this in terms of the system.

File: cves/CVE-2014-1702.yml [hunk 4]
```diff
@@ -48,129 +51,131 @@ unit_tested:
...
-
-    For the "fix" answer below, check if the fix for the vulnerability involves
-    adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+    For the fix answer below, check if the fix for the vulnerability involves
+    adding or improving an automated test to ensure this doesnt happen again.
+  answer: |
+    The original code was not united tested and neither is the fix. The
```
Comment (jonobrien) [inline cves/CVE-2014-1702.yml]: "unit" ?

File: cves/CVE-2014-1702.yml [hunk 5]
```diff
@@ -48,129 +51,131 @@ unit_tested:
...
   secure_by_default:
-    applies: 
+    applies: false
     note: 
   yagni:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      You arent going to need it seems to fit this vulnerability. This vulnerability was
```
Comment (jonobrien) [inline cves/CVE-2014-1702.yml]: "aren't" ?

File: cves/CVE-2014-1702.yml (hunk 4 above)
Comment (smcintosh881) [inline cves/CVE-2014-1702.yml]: Good catch!

File: cves/CVE-2014-1702.yml (hunk 2 above)
Comment (smcintosh881) [inline cves/CVE-2014-1702.yml]: I'm not really sure what you mean by this. The vulnerability leaves the system open as threads are not properly shutdown and attackers can point them to execute different code. What would help get this idea across to the reader?

File: cves/CVE-2014-1702.yml [hunk 6]
```diff
@@ -15,10 +15,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
-bounty:
+description: |
+  This vulnerability is of the free after use nature. When a task is running on a thread,
+  and that task ends the thread is being shutdown. This vulnerability could allow an
+  attacker to creat an denial of service. They leverage the improper handling of how
```
Comment (jonobrien) [inline cves/CVE-2014-1702.yml]: "create"

File: cves/CVE-2014-1702.yml (hunk 2 above)
Comment (jonobrien) [inline cves/CVE-2014-1702.yml]: >  The vulnerability leaves the system open as threads are not properly shutdown and attackers can point them to execute different code

This sounds good. Something that explains the actual results of the vulnerability besides DOS, unless DOS is the only thing relevant besides arbitrary code execution.

Comment (smcintosh881) [inline cves/CVE-2014-1702.yml]: It seems that the biggest concern for this specific vulnerability is DOS. This is the main negative effect discussed about this vulnerability.

**Pull request merged**